inc/lift/Request.h

```diff
@@ -349,6 +433,45 @@ class Request {
     /// libuv will call this function when the AddRequest() function is called.
     friend auto requests_accept_async(
         uv_async_t* handle) -> void;
+    
+    friend auto on_response_wait_time_expired_callback(uv_timer_t* handle) -> void;
+};
+
+/**
+ * Class wrapping information used by EventLoop to "time out" requests that have not
+ * received a response within the expected wait time.
+ */
+class ResponseWaitTimeWrapper
+{
+private:
+    struct Data
+    {
+        /**
+         * Represents the time point when the corresponding request should be timed out.
+         * Using uint64_t for consistency since uv_now returns time points as uint64_t.
+         */
+        uint64_t m_timeout_time;
+        /// Reference to the Request associated with this wrapper.
+        std::shared_ptr<SharedRequest>* m_shared_request_ptr_pointer;
+    } m_data;
+public:
+    ResponseWaitTimeWrapper(uint64_t timeout_time, std::shared_ptr<SharedRequest>* request) : m_data{timeout_time, request} {}
+    
+    /**
+     * @return Reference to the const Data struct associated with this wrapper.
+     */
+    [[nodiscard]]
+    auto GetData() const -> const Data& { return m_data; }
```

Comment:
Seems odd that this returns a ref to a private struct

Reply:
Good point, the struct definition is public now